src/tvuikit.cs

```diff
@@ -290,4 +300,166 @@ interface TVCollectionViewFullScreenLayout {
 		[Export ("transitioningToCenterIndexPath")]
 		bool TransitioningToCenterIndexPath { [Bind ("isTransitioningToCenterIndexPath")] get; }
 	}
+
+	[TV (15,0), NoWatch, NoMac, NoiOS]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface TVMediaItemContentConfiguration : UIContentConfiguration, NSSecureCoding {
+
+		[Static]
+		[Export ("wideCellConfiguration")]
+		TVMediaItemContentConfiguration CreateWideCell ();
+
+		[NullAllowed, Export ("image", ArgumentSemantic.Strong)]
+		UIImage Image { get; set; }
+
+		[NullAllowed, Export ("text")]
+		string Text { get; set; }
+
+		[Export ("textProperties")]
+		TVMediaItemContentTextProperties TextProperties { get; }
+
+		[NullAllowed, Export ("secondaryText")]
+		string SecondaryText { get; set; }
+
+		[Export ("secondaryTextProperties")]
+		TVMediaItemContentTextProperties SecondaryTextProperties { get; }
+
+		[Export ("playbackProgress")]
+		float PlaybackProgress { get; set; }
+
+		[NullAllowed, Export ("badgeText")]
+		string BadgeText { get; set; }
+
+		[Export ("badgeProperties", ArgumentSemantic.Copy)]
+		TVMediaItemContentBadgeProperties BadgeProperties { get; set; }
+
+		[NullAllowed, Export ("overlayView", ArgumentSemantic.Strong)]
+		UIView OverlayView { get; set; }
+	}
+
+	[TV (15,0), NoWatch, NoMac, NoiOS]
+	[BaseType (typeof (UIView))]
+	[DisableDefaultCtor]
+	interface TVMediaItemContentView : UIContentView {
+
+		[Export ("initWithConfiguration:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (TVMediaItemContentConfiguration configuration);
+
+		// Conflicts with IUIContentView.Configuration property unfortunately
```

Comment:
hmm...

`UIContentConfiguration` is a protocol and `TVMediaItemContentConfiguration` conforms to it. So the `IUIContentView.Configuration` property is usable...

The setter is usable without any issue. However the getter is not optimal but...

It might be better, and more general, to add an helper/extension method like `GetConfiguration<T> ()` that would (internally) call the getter and return an instance of the request subclass of `IUIContentConfiguration`

Reply:
Right, but is it really worthy creating the extension ‘ GetConfiguration<T> ()’ if you only have two places where it is used? I think by  just keeping the sealed GetConfiguration() on each subclass is a bit better since it avoids user guess work and any generics involvement

Reply:
I think so as it can be done, once, as an extension method on the interface (for the protocol) without requiring each type to add new API (hopefully with a consistent name over time) and with additional `[Export]` (that the registrar has to process).